test/ng/compileSpec.js

```diff
@@ -12111,6 +12142,890 @@ describe('$compile', function() {
+
+            // As a RESOURCE URL
+            element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+            $rootScope.testUrl = $sce.trustAsResourceUrl('javascript:foo()');
+            $rootScope.$digest();
+            expect(element.find(tag).prop('src')).toEqual('javascript:foo()');
+          }));
+        });
+      });
+    }
+
+    describe('img[src] sanitization', function() {
+
+      it('should accept trusted values', inject(function($rootScope, $compile, $sce) {
+        element = $compile('<img ng-prop-src="testUrl"></img>')($rootScope);
+        // Some browsers complain if you try to write `javascript:` into an `img[src]`
+        // So for the test use something different
+        $rootScope.testUrl = $sce.trustAsMediaUrl('someuntrustedthing:foo();');
+        $rootScope.$digest();
+        expect(element.prop('src')).toEqual('someuntrustedthing:foo();');
+      }));
+
+      it('should use $$sanitizeUri', function() {
+        var $$sanitizeUri = jasmine.createSpy('$$sanitizeUri');
+        module(function($provide) {
+          $provide.value('$$sanitizeUri', $$sanitizeUri);
+        });
+        inject(function($compile, $rootScope) {
+          element = $compile('<img ng-prop-src="testUrl"></img>')($rootScope);
+          $rootScope.testUrl = 'someUrl';
+
+          $$sanitizeUri.and.returnValue('someSanitizedUrl');
+          $rootScope.$apply();
+          expect(element.prop('src')).toMatch(/^http:\/\/.*\/someSanitizedUrl$/);
+          expect($$sanitizeUri).toHaveBeenCalledWith($rootScope.testUrl, true);
+        });
+      });
+
+      it('should not use $$sanitizeUri with trusted values', function() {
+        var $$sanitizeUri = jasmine.createSpy('$$sanitizeUri').and.throwError('Should not have been called');
+        module(function($provide) {
+          $provide.value('$$sanitizeUri', $$sanitizeUri);
+        });
+        inject(function($compile, $rootScope, $sce) {
+          element = $compile('<img ng-prop-src="testUrl"></img>')($rootScope);
+          // Assigning javascript:foo to src makes at least IE9-11 complain, so use another
+          // protocol name.
+          $rootScope.testUrl = $sce.trustAsMediaUrl('untrusted:foo();');
+          $rootScope.$apply();
+          expect(element.prop('src')).toBe('untrusted:foo();');
+        });
+      });
+    });
+
+    ['img', 'source'].forEach(function(srcsetElement) {
+      describe(srcsetElement + '[srcset] sanitization', function() {
+        it('should not error if srcset is undefined', inject(function($compile, $rootScope) {
+          element = $compile('<' + srcsetElement + ' ng-prop-srcset="testUrl"></' + srcsetElement + '>')($rootScope);
+          // Set srcset to a value
+          $rootScope.testUrl = 'http://example.com/';
+          $rootScope.$digest();
+          expect(element.prop('srcset')).toBe('http://example.com/');
+
+          // Now set it to undefined
```

Comment:
Undefined you say... 🤔

Reply:
The problems with copy pasta...

Reply:
Will also update the test description because actually it should error when set to undefined, because the property converts that to the `"undefined"` string value.